src/cmap/auth/mongo_credentials.ts

```diff
@@ -61,16 +61,19 @@ export class MongoCredentials {
     this.mechanismProperties = options.mechanismProperties || {};
 
     if (this.mechanism.match(/MONGODB-AWS/i)) {
-      if (this.username == null && process.env.AWS_ACCESS_KEY_ID) {
+      if (!this.username && process.env.AWS_ACCESS_KEY_ID) {
         this.username = process.env.AWS_ACCESS_KEY_ID;
       }
 
-      if (this.password == null && process.env.AWS_SECRET_ACCESS_KEY) {
+      if (!this.password && process.env.AWS_SECRET_ACCESS_KEY) {
         this.password = process.env.AWS_SECRET_ACCESS_KEY;
       }
 
-      if (this.mechanismProperties.AWS_SESSION_TOKEN == null && process.env.AWS_SESSION_TOKEN) {
-        this.mechanismProperties.AWS_SESSION_TOKEN = process.env.AWS_SESSION_TOKEN;
+      if (process.env.AWS_SESSION_TOKEN) {
```

Comment:
I mean … does this not seem like a spec bug? I would really wonder what their reasons for this are, especially as the description for the credentials `AWS_SESSION_TOKEN` property also states:

> ***AWS_SESSION_TOKEN***
>   Drivers MUST allow the user to specify an AWS session token for authentication with temporary credentials.

Reply:
Yeah the spec seems to contradict itself in that section, reading to me like URI takes precedence, but if `AWS_SESSION_TOKEN` is there it takes precedence. I think maybe a SPEC ticket to clear that up?

I think for the purposes of this PR though - this looks good to me.

Reply:
I just tried opening a SPEC ticket - maybe I lack permissions to do that? Can anyone of you two check if they could open that up? 🤔

Reply:
I think [DRIVERS-1405](https://jira.mongodb.org/browse/DRIVERS-1405) relates to this.

Reply:
That first paragraph quoted in that ticket is also relevant:

> The order in which Drivers MUST search for credentials is:
>
> #. The URI
> #. Environment variables
> #. The ECS endpoint if ``AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`` is set. Otherwise, the EC2 endpoint.

I don’t know to what degree options as opposed to URI parameters are a Node.js-driver-specific concept, but I would also read this as saying that the environment variable should not take precedence over user-provided options.

I would undo this particular change here.

Reply:
After talking to @nbbeeken I've now changed that so the explicit authMechanismProperties take precedence similar as username and password. I'll therefore resolve that conversation.